azkaban-web-server/src/test/java/azkaban/restli/DeployProjectTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.restli;
+
+import static org.junit.Assert.assertEquals;
+
+import azkaban.project.validator.ValidationReport;
+
+import com.linkedin.restli.common.HttpStatus;
+import com.linkedin.restli.server.RestLiServiceException;
+
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Test;
+
+/**
+ * Test response to deploy with either warn or error reports.
+ * If a report has any errors, a RestLiServiceException should be thrown.
+ * Warnings should not elicit an exception.
+ * If an exception does get raised, it should carry an HTTP 400 status
+ */
+
+public class DeployProjectTest {
+  @Test
+  public void testWarnedDeploy() {
+    ProjectManagerResource resource = new ProjectManagerResource();
+    Map<String, ValidationReport> reports = new LinkedHashMap<String, ValidationReport>();
+    for (int i = 0; i < 3; i++) {
+      Set<String> warnMsgs = new HashSet<String>();
+      ValidationReport warnRpt = new ValidationReport();
```

Comment:
Nice job with the tests!

While this works. Consider having a simple method for created the kind of mock objects that you need for all your testing.

All your tests require creating ValidationReport objects with warn / error messages so why not create a simple method that gives you that with ease. 
Example: 
`
ValidationReport validationWarn(String msg) {
  // return new created ValidationReport object with the msg. 
}
  ValidationReport validationError(String msg);
`

This results in lesser code duplication plus makes it more maintainable in case you need to add some other field later on.

Make your tests as simple / small as possible and extract such constructs out so that adding new cases is a very minimal effort.